dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -1807,6 +1781,49 @@ private boolean deleteContentlets(List<Contentlet> contentlets, User user,
         return noErrors;
     }
 
+    /**
+     * Verifies if a page is being used as a detail page for any content type
+     * @param user
+     * @param c
+     * @throws DotDataException
+     * @throws LanguageException
+     */
+    private void validateRelatedContentType(User user, Contentlet c)
+            throws DotDataException{
+        ContentTypeAPI contentTypeAPI = APILocator.getContentTypeAPI(user);
+        List<ContentType> relatedContentTypes = contentTypeAPI.search("page_detail='" + c.getIdentifier() + "'");
+        HTMLPageAssetAPI htmlPageAssetAPI = APILocator.getHTMLPageAssetAPI();
+        String uri = htmlPageAssetAPI.fromContentlet(c).getURI();
+        //Verifies if the page is related to any content type
+        if (relatedContentTypes != null && !relatedContentTypes.isEmpty()){
+
+            //Unlinking url map and detail page
+            relatedContentTypes.forEach((ContentType contentType) -> {
+                try {
+                    contentTypeAPI.unlinkPageFromContentType(contentType);
+                } catch (DotSecurityException | DotDataException e) {
+                    throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)